olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h

```diff
@@ -119,6 +119,11 @@ class DATASERVICE_READ_API CatalogClient final {
   /**
    * @brief Gets the catalog version asynchronously.
    *
+   * @note For CacheOnly requests please be sure to specify existing version as
+   * a `StartVersion` in the request so that version could be stored as a new
+   * latest one in case of absence already cached version or when cached version
+   * is less then a new one.
```

Comment:
```suggestion
   * @note In case you call this API with `FetchOptions::CacheOnly` and a valid version in `CatalogVersionRequest::WithStartVersion()`, i.e. >= 0, then please make sure that the provided version is a existing catalog version as it will be written for later use to the cache as latest version in the following cases:
   * - There is no latest version yet written to cache.
   * - The latest version written to cache is less then the provided version.  
```